cmd/juju/caas/aks.go

```diff
@@ -0,0 +1,287 @@
+// Copyright 2019 Canonical Ltd.
+// Licensed under the AGPLv3, see LICENCE file for details.
+
+package caas
+
+import (
+	"encoding/json"
+	"fmt"
+	"io"
+	"os"
+
+	"github.com/juju/cmd"
+	"github.com/juju/collections/set"
+	"github.com/juju/errors"
+
+	"github.com/juju/juju/caas/kubernetes/clientconfig"
+	"github.com/juju/juju/cmd/juju/interact"
+)
+
+type aks struct {
+	CommandRunner
+	azExecName string
+}
+
+func newAKSCluster() k8sCluster {
+	return &aks{CommandRunner: &defaultRunner{}}
+}
+
+func (a *aks) cloud() string {
+	return "aks"
```

Comment:
This has to be "azure", ie a recognised Juju cloud type.
I'm surprised this passed manual testing. I suspect the missing code to set cloud/region explains it.

Reply:
You where right about that Ian.